### ISSUE TITLE
Can't merge pod_target_xcconfig for pod targets: ["GIF", "WebP"]. Singular build setting USER_HEADER_SEARCH_PATHS has different values.

### ISSUE THREAD

**Issue by cj1024** [SDWebImage/SDWebImage#2619]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios
 Platform Version        | 12.0
 SDWebImage Version      | 4.4.5
 Integration Method      | cocoapods
 Xcode Version           | Xcode 10
 Repro rate              | all the time (100%)
 Repro with our demo prj | The demo doesn't integrate with cocoapods

### Issue Description and Steps

After installing both `SDWebImage/WebP` and `SDWebImage/GIF` with `pod install`, you will get warnings like `[!] Can't merge pod_target_xcconfig for pod targets: ["GIF", "WebP"]. Singular build setting USER_HEADER_SEARCH_PATHS has different values.`. 
Perhaps you should use `pod_target_xcconfig` instead of `xcconfig` in the podspec file.

**Comment by dreampiggy** [comment]
It's CocoaPods's bug (from v1.6.0-beta), there are issue about it via https://github.com/CocoaPods/CocoaPods/issues/8451. And been fixed via https://github.com/CocoaPods/CocoaPods/pull/8452

**Comment by pranavkth** [comment]
I have also encountered this issue and this is resolved by adding 
$(SRCROOT)/libwebp/src and $(inherited) 
in SDWebImageFramework in Pods target under User Header Search Paths in build settings.

But I need to do this manually everytime I make a pod install.
Please resolve this asap team.

Using cocoaPods version 1.6

**Comment by dreampiggy** [comment]
> Please resolve this asap team.

Actually, it's a CocoaPods's bug, but not the way for framework author to solve. The podspec syntax here, use the standard `pod_target_xcconfig`, with the correct `$(inherited)` synxtax for user header search path xcconfig.

The issue have already been fixed by CocoaPods, it's available after next release, or you can have a try with the latest master branch of CocoaPods.

**Comment by dreampiggy** [comment]
OK. My fault, I found that SDWebImage using `spec.xcconfig` syntax ? It's a little strange, let me have a try with `pod_target_xcconfig` instead.

**Comment by dreampiggy** [comment]
Seems even I change the `spec.xcconfig` into `spec.pod_target_xcconfig`, the warning on CocoaPods 1.6.0 still exist.

But actually, those user header search path have no any meaning. I'll have a check again, maybe it's better to remove those build settings.

This podspec can work:

```ruby
  s.subspec 'GIF' do |gif|
    gif.ios.deployment_target = '7.0'
    gif.source_files = 'SDWebImage/FLAnimatedImage/*.{h,m}'
    gif.dependency 'SDWebImage/Core'
    gif.dependency 'FLAnimatedImage', '~> 1.0'
  end

  s.subspec 'WebP' do |webp|
    webp.source_files = 'SDWebImage/UIImage+WebP.{h,m}', 'SDWebImage/SDWebImageWebPCoder.{h,m}'
    webp.pod_target_xcconfig = { 
      'GCC_PREPROCESSOR_DEFINITIONS' => '$(inherited) SD_WEBP=1',
    }
    webp.watchos.pod_target_xcconfig = {
      'GCC_PREPROCESSOR_DEFINITIONS' => '$(inherited) SD_WEBP=1 WEBP_USE_INTRINSICS=1',
    }
    webp.dependency 'SDWebImage/Core'
    webp.dependency 'libwebp', '~> 0.5'
  end
```

**Comment by dreampiggy** [comment]
@cj1024 Should be fixed via #2622.

**Comment by cj1024** [comment]
> @cj1024 Should be fixed via #2622.

That's great
